ExpenseTracker/ExpenseTracker/Controllers/HomeController.cs

```diff
@@ -26,11 +26,17 @@ public IActionResult Index()
     }
 
     [Route("Home/Error")]
-    public IActionResult Error()
+    public IActionResult Error(int? statusCode = 500)
     {
-        return View();
+        if (statusCode == 404)
+        {
+            return View("NotFound");
+        }
+
+        return View("Error");
```

Comment:
Have you used pattern matching? I Think it would be perfect fit here.

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#compare-discrete-values

Reply:
Fixed. Thanks